unreleased/multi-package.md

```diff
@@ -0,0 +1,462 @@
+├── package-Model
+└── package-Tests
+```
+
+In the newly created `package-Tests/daml.yaml`, we remove the `sandbox-options` and add a dependency on the DAR for `Logic`.
+
+```bash
+data-dependencies:
+- ../package-Logic/.daml/dist/package-Logic-1.0.0.dar
+```
+
+Finally, we'll want to tie the new package into our project to take advantage of the multi-package feature - we do this by adding `package-Testing` to the `multi-package.yaml` in the same way Logic and Model have been added.
+
+`multi-package.yaml`
+```diff
+  packages:
+  - ./package-Logic
+  - ./package-Model
++ - ./package-Tests
+```
+
+#### Running Multi-Package Tests
+
+Assuming we then write our tests, we can build our test package in one of two ways:
+
+- Run `daml build` from the `package-Testing` directory to build the Tests package and its dependency Logic package.
+- Run `daml build --all` from the root of the project to build all three packages in our project, including the Tests package.
+
+Once we have built the package, we can run `daml test` from the `package-Testing` directory to run the up-to-date tests.
+
+### Behaviour for Packages Not Listed In multi-package.yaml
+
+Packages that are not listed in the `multi-package.yaml` file will not be automatically recompiled, regardless of whether their source is available in the project's source tree.
+
+For example, say we add a vendor library to our source tree. We build the vendor library from source in isolation, and refer to its `vendor-library.dar` as a data-dependency from within `package-Logic/daml.yaml`.
+
+```
+> tree
+.
+├── multi-package.yaml
+├── package-Logic
+│   ├── daml/...
+│   └── daml.yaml
+├── package-Model
+│   ├── daml/...
+│   ├── daml/.dist/package-Model-1.0.0.dar
+│   └── daml.yaml
+└── vendor-library
+    ├── daml/...
+    ├── daml/.dist/vendor-library-1.0.0.dar
+    └── daml.yaml
+```
+
+If we do not update `multi-package.yaml`, builds of package Logic will **not** try to rebuild `vendor-library.dar`, even though its source and `vendor-library/daml.yaml` are clearly available.
+
+```bash=
+> # ... make some changes to vendor-library ...
+> cd ./package-Logic/ # Navigate to package-Logic
+> daml build # Build package Logic
+...
+Building "package-Logic"... # We DO NOT rebuild vendor-library
+Done.
+```
+
+This allows users to distinguish between packages that are part of a given project and are not, without risking recompilation of a vendored package whose upstream they don't control.
+
+## Working with Multiple Projects
+
+In some cases, we have nested projects. For example, say we have two separate GitHub repositories, `application` and `library`, in which we work and change the source regularly. The application repository depends on the library-repository by pointing at DAR files within the library repository.
+
+```
+.
+├── application-repository
+│   ├── .git
+│   ├── application-logic
+│   │   └── daml.yaml
+│   ├── application-tests
+│   │   └── daml.yaml
+│   └── multi-package.yaml
```

Comment:
Would it make sense that all of the directory trees have the multi-package.yaml at the top of the list, below the .git directory?  This is easier to notice.

Reply:
fixed